HHS/Head-Start-TTADP
# [TTAHUB-3036] Filter out TR goals from similarity api

## Description of change

We want to exclude TR goals for the similarity API. This change does two things:
- Prevents goals with a createdVia of 'tr' from coming back via the API.
- Delete's any TR goals that currently have merge suggestions but are NOT merged.

As of writing this there are currently **no TR goals that have been merged**. 

## How to test

- Review the code.
- Run the **test.http** test called '### given a text, compute similarities for this recipient' for a TR goal and recipient (nothing should come back).
- Run the same test in the above step for a valid createdVia it should return results.
- Run the migration make sure we no longer have any 'GoalSimilarityGroups' or 'GoalSimilarityGroupGoals' for non-merged TR goals.

## Issue(s)

* https://jira.acf.gov/browse/TTAHUB-3036


## Checklists

### Every PR



### Before merge to main


### Production Deploy


### After merge/deploy
